src/mono/mono.proj

```diff
@@ -28,15 +28,24 @@
     <EscapedQuoteW Condition="'$(OS)' == 'Windows_NT'">\&quot;</EscapedQuoteW>
     <PythonCmd Condition="'$(OS)' != 'Windows_NT'">python3</PythonCmd>
     <PythonCmd Condition="'$(OS)' == 'Windows_NT'">python</PythonCmd>
-    <CoreClrFileName Condition="'$(TargetsWindows)' == 'true'">coreclr.dll</CoreClrFileName>
-    <CoreClrFileName Condition="'$(TargetsOSX)' == 'true'">libcoreclr.dylib</CoreClrFileName>
-    <CoreClrFileName Condition="'$(TargetsMacCatalyst)' == 'true'">libcoreclr.dylib</CoreClrFileName>
-    <CoreClrFileName Condition="'$(CoreClrFileName)' == ''">libcoreclr.so</CoreClrFileName>
-    <MonoFileName Condition="'$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">libmonosgen-2.0.dylib</MonoFileName>
-    <MonoFileName Condition="'$(TargetsAndroid)' == 'true'">libmonosgen-2.0.so</MonoFileName>
-    <MonoFileName Condition="'$(TargetsBrowser)' == 'true'">libmonosgen-2.0.a</MonoFileName>
+    <CoreClrLibName>coreclr</CoreClrLibName>
+    <CoreClrFileName Condition="'$(TargetsWindows)' == 'true'">$(CoreClrLibName).dll</CoreClrFileName>
+    <CoreClrFileName Condition="'$(TargetsOSX)' == 'true'">lib$(CoreClrLibName).dylib</CoreClrFileName>
+    <CoreClrFileName Condition="'$(TargetsMacCatalyst)' == 'true'">lib$(CoreClrLibName).dylib</CoreClrFileName>
+    <CoreClrFileName Condition="'$(CoreClrFileName)' == ''">lib$(CoreClrLibName).so</CoreClrFileName>
+    <MonoLibName>monosgen-2.0</MonoLibName>
+    <MonoSharedLibName Condition="'$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true' or '$(TargetsAndroid)' == 'true' or '$(TargetsBrowser)' == 'true'">$(MonoLibName)</MonoSharedLibName>
+    <MonoSharedLibName Condition="'$(MonoSharedLibName)' == ''">$(CoreClrLibName)</MonoSharedLibName>
+    <MonoFileName Condition="'$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">lib$(MonoSharedLibName).dylib</MonoFileName>
+    <MonoFileName Condition="'$(TargetsAndroid)' == 'true'">lib$(MonoSharedLibName).so</MonoFileName>
+    <MonoFileName Condition="'$(TargetsBrowser)' == 'true'">lib$(MonoLibName).a</MonoFileName>
     <MonoFileName Condition="'$(MonoFileName)' == ''">$(CoreClrFileName)</MonoFileName>
-    <MonoStaticFileName Condition="'$(TargetsMacCatalyst)' == 'true' or '$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true' or '$(TargetsAndroid)' == 'true'">libmonosgen-2.0.a</MonoStaticFileName>
+    <MonoStaticFileName Condition="'$(TargetsMacCatalyst)' == 'true' or '$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true' or '$(TargetsAndroid)' == 'true'">lib$(MonoLibName).a</MonoStaticFileName>
+    <SharedNativeLibrarySuffix Condition="'$(TargetsOSX)' == 'true' or '$(TargetsMacCatalyst)' == 'true' or '$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">.dylib</SharedNativeLibrarySuffix>
```

Comment:
I think that SharedNativeLibrarySuffix is the same of LibExt, maybe we could use only one?

Reply:
Need to add a couple platforms currently not in more to the LibExt and then we could switch.

Reply:
Don't worry, I can use SharedNativeLibrarySuffix when I do my next PR.

Reply:
Since I did some other changes related to review feedback I reworked this as well.